src/pal/src/synchmgr/synchmanager.cpp

```diff
@@ -1835,16 +1878,36 @@ namespace CorUnix
             palErr = pSynchManager->ReadCmdFromProcessPipe(iPollTimeout, 
                                                            &swcCmd,
                                                            &shridMarshaledData,
-                                                           &dwData);            
+                                                           &dwData);
             if (NO_ERROR != palErr)
             {
                 ERROR("Received error %x from ReadCmdFromProcessPipe()\n",
-                      palErr);                
+                      palErr);
                 continue;
             }
             switch (swcCmd)
             {
-                case SynchWorkerCmdNop:                    
+                case SynchWorkerCmdTerminationRequest:
+                    // This worker thread is being asked to initiate process termination
+
+                    HANDLE hShutdownThread;
+                    palErr = InternalCreateThread(pthrWorker,
+                                      NULL,
+                                      0,
+                                      &TerminationRequestHandlingRoutine,
+                                      NULL,
+                                      0,
+                                      PalWorkerThread,
+                                      NULL,
+                                      &hShutdownThread);
+
+                    if (NO_ERROR != palErr)
+                    {
+                        ERROR("Unable to create worker thread\n");
+                    }
```

Comment:
It may be nice to close thread handle. I know it is not strictly necessary for the shutdown case because of the process is likely go away, but it is a good hygiene.